src/product/entities/product.entity.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  Column,
+  Entity,
+  PrimaryGeneratedColumn,
+  CreateDateColumn,
+} from 'typeorm';
+
+import { ProductState } from '../types/product';
+
+@Entity('products')
+export class Product {
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
+
+  @Column('text', { unique: true })
+  name: string;
+
+  @Column('text')
+  description: string;
+
+  @Column('numeric', { name: 'num_available_items' })
+  numAvailableItems: number;
+
+  @Column('numeric', { name: 'unit_price' })
+  unitPrice: number;
+
+  @Column('text')
+  state: ProductState;
```

Comment:
You could make the 'state' column an enum using this syntax:
```suggestion
  @Column({
        type: "enum",
        enum: ProductState,
    })
  state: ProductState;
```

Reply:
nice, I didn't know this existed in the first place . Thanks